sda-commons-server-opentelemetry/docs/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,148 @@
+# Migration Guide
+
+## Java Version
+ You must use at least Java 8 to use OpenTelemetry. [More info](https://opentelemetry.io/docs/migration/opentracing/#language-version-support)
+
+## Replace OpenTracing dependencies to OpenTelemetry dependencies
+
+Before:
+```gradle
+testImplementation 'io.opentracing:opentracing-mock'
+```
+
+After:
+```gradle
+testImplementation 'io.opentelemetry:opentelemetry-sdk-testing'
+```
+
+## Executors need to be instrumented to follow traces when parts of the request are handled asynchronously.
+
+Before: 
+```java
+return new AsyncEmailSendManager(
+  new InMemoryAsyncTaskRepository<>(limits.getMaxAttempts()),
+  environment
+      .lifecycle()
+      .executorService("email-sender-%d")
+      .minThreads(1)
+      .maxThreads(limits.getMaxParallel())
+      .build(),
+```
+
+After:
+
+```java
+return new AsyncEmailSendManager(
+    new InMemoryAsyncTaskRepository<>(limits.getMaxAttempts()),
+    Context.taskWrapping(
+      environment
+          .lifecycle()
+          .executorService("email-sender-%d")
+          .minThreads(1)
+          .maxThreads(limits.getMaxParallel())
+          .build()),
+```
+
+## Environment variables
+
+To disable tracing, you must set the variable `TRACING_DISABLED` to `true`. The legacy Jaeger environment variables are still supported, but they will be removed in later versions.
+
+Before:
+```properties
+JAEGER_SAMPLER_TYPE=const
+JAEGER_SAMPLER_TYPE=0
+```
+
+After:
+```properties
+TRACING_DISABLED=true
+```
+
+## New environment variables
+In order to configure Open Telemetry, you have to set some environment variables:
+
+| Name                          | Default value                                 | Description                                                                      |
+|-------------------------------|-----------------------------------------------|----------------------------------------------------------------------------------|
+| OTEL_PROPAGATORS              | jaeger,b3,tracecontext,baggage                | Propagators to be used as a comma-separated list                                 |
+| OTEL_SERVICE_NAME             | value from env `JAEGER_SERVICE_NAME` (if set) | The service name that will appear on tracing                                     |
```

Comment:
Nice to know. Then we don't need the one setting we have in the test deployment for a real upgrade. However, in the test deployment it helps to separate the traces from the regular deployment of the service.